### PR TITLE
Upgrade QRGen to v2.5.0 in wallettemplate

### PR DESCRIPTION
**Pull request by msgilligan**
* QRGen 2.5.0 (eliminates split packages - also requires use of jitpack.io to get latest build)

### CODE REVIEW

**Comment by schildbach** [comment]
I'm skeptical that adding another repo is worth the new QRGen version.

If you separate the commits I'd merge just the EasyBind update for now.

**Comment by msgilligan** [comment]
It's the upgraded QRGen version that I really want. It fixes a problem with split packages that breaks Java Modules -- which is something I'm working on for a future pull request. Notice that I added the `jitpack.io` repo only to the build for `wallettemplate` which will only run with Java 11 is installed.

**Comment by msgilligan** [comment]
I'll go ahead and create a separate PR for the **EasyBind** update and make this one for **QRGen** only.

**Comment by msgilligan** [comment]
OK, I force-pushed a change to this PR so it has the **QRGen** v2.5.0 upgrade only. I also submitted PR #1770 to contain just the **EasyBind** update.

**Comment by schildbach** [comment]
What version of QRGen fixes the issue with split packages? I noticed there _are_ more recent QRGen releases on JCenter, just not the most recent ones.

Also I wonder if we actually need QRGen. Creating QR-codes with ZXing is a one-liner, and it seems as QRGen is using ZXing behind the scenes anyway.

**Comment by schildbach** [comment]
Just for clarification: I'm not against Jitpack, but I see the disadvantages of using multiple repos at the same time. I would suggest if we switch, we switch with the entire project. And since this has implications on the consumers of bitcoinj too I think such a change should be openly discussed on the mailing list first.

**Comment by msgilligan** [comment]
Yeah, I'm not a fan of using jitpack for what should be released JARs. I definitely don't want to switch the entire project.

I like your idea of using ZXing directly and moving the QRGen dependency. I'll make that change in the next few days and force-push.

**Comment by msgilligan** [comment]
> I like your idea of using ZXing directly and moving the QRGen dependency. I'll make that change in the next few days

Done!

> and force-push

Not done (because a local pop/merge interfered.)

If the cumulative diff looks acceptable to you, I'll do a squash and force-push.

File: wallettemplate/src/main/java/wallettemplate/controls/ClickableBitcoinAddress.java [hunk 1]
```diff
@@ -153,4 +150,38 @@ protected void showQRCode(MouseEvent event) {
...
+    /**
+     * Create a PNG-format image byte string from a Bitcoin URI
+     * @param uri Bitcoin URI
+     * @return PNG-format bytes
+     */
+    private static byte[] qrCodePNGBytes(String uri) {
+        Writer qrWriter = new QRCodeWriter();
+        BitMatrix matrix;
+        try {
+            matrix = qrWriter.encode(uri, com.google.zxing.BarcodeFormat.QR_CODE, 320, 240);
```
Comment (schildbach) [inline wallettemplate/src/main/java/wallettemplate/controls/ClickableBitcoinAddress.java]: Maybe import BarcodeFormat?

Comment (msgilligan) [reply wallettemplate/src/main/java/wallettemplate/controls/ClickableBitcoinAddress.java]: Good idea, will do.

**Comment by msgilligan** [comment]
> I wonder do I understand right that the QR code will be PNG-encoded and decoded immediately after that? 

Yeah, see Mike's comment. I decided not to try to fix that in this commit.

> I don't know about the capabilities of JavaSE/JFX, but can't the QR code be rendered directly into a black/white bitmap that is used for displaying?

There must be a way. And we may want to switch libraries anyway because ZXing says it is in "maintenance mode" and has a problem with [Split Packages](https://github.com/zxing/zxing/issues/1153) that is making a modular implementation of `wallettemplate` more difficult.

Maybe we should look at [nayuki/QR-Code-generator](https://github.com/nayuki/QR-Code-generator) and the code in (rejected) [Issue 33: Pure JavaFX WriteImage code](https://github.com/nayuki/QR-Code-generator/issues/33).

**Comment by nayuki** [comment]
I would be happy to see my QR Code generator library be used in bitcoinj. Is there anything I can help you with?

**Comment by schildbach** [comment]
Merged.

Regarding switching out ZXing for QR Code generator: Encoding is one part, decoding would be the other. Apparently wallet-template doesn't have a QR code scanner, but a fully fledged wallet would need to have that. So I'm not so much in favour of switching, especially because ZXing works very well.

Maintenance mode is also not a problem for me, as long as it's still maintained.

The problem with split packages could be worked around by only using the ZXing core package. It has a generic API that returns a "BitMatrix" that you can easily copy into your own bitmap. See https://github.com/bitcoin-wallet/bitcoin-wallet/blob/master/wallet/src/de/schildbach/wallet/util/Qr.java#L54 for example.

**Comment by schildbach** [comment]
I just saw this PR uses BitMatrix already. But why do we need the ZXing-JavaSE package then?

**Comment by msgilligan** [comment]
> But why do we need the ZXing-JavaSE package then?

You're right. The only thing we're using from the ZXing-JavaSE package is the PNG conversion. The solution is to remove ZXing-JavaSE and convert directly to a JavaFX `Image`. This will kill two birds with one stone.

I'll submit a new PR with this change.

**Comment by msgilligan** [comment]
Thanks @nayuki your QR Code generator library looks awesome, but we're sticking with ZXing for the time being.

**Pull request closed without merging**